src/node/gain.rs

```diff
@@ -94,16 +94,18 @@ impl AudioProcessor for GainRenderer {
         let input = &inputs[0];
         let output = &mut outputs[0];
 
-        let gain_values = params.get(&self.gain);
-
         *output = input.clone();
 
-        output.modify_channels(|channel| {
-            channel
-                .iter_mut()
-                .zip(gain_values.iter())
-                .for_each(|(value, g)| *value *= g)
-        });
+        if !input.is_silent() {
+            let gain_values = params.get(&self.gain);
+
+            output.modify_channels(|channel| {
+                channel
+                    .iter_mut()
+                    .zip(gain_values.iter())
+                    .for_each(|(value, g)| *value *= g)
+            });
+        }
```

Comment:
You still should fill the output with silence here, otherwise the outputs are unspecified (reused from previous render quantum)

Reply:
hum I don't understand, has `*output = input.clone();` (line 97) we are just propagating forward the silent input, or did I miss something?

Reply:
You are right, sorry I missed that part